lib/api/apiUtils/object/coldStorage.js

```diff
@@ -58,6 +58,7 @@ function setArchiveInfoHeaders(objMD) {
         if (objMD.archive.restoreCompletedAt) {
             headers['x-amz-scal-restore-completed-at'] = new Date(objMD.archive.restoreCompletedAt).toUTCString();
             headers['x-amz-scal-restore-will-expire-at'] = new Date(objMD.archive.restoreWillExpireAt).toUTCString();
+            headers['x-amz-scal-restore-etag'] = objMD['x-amz-restore'].etag;
```

Comment:
I think in this code path that the header x-amz-restore might not always be set, can we ensure it exists before access this property (same applies to other relevant places)

For example in the build I see this, from the quota tests:
```
{"name":"S3","time":1727859015536,"error":"Cannot read properties of undefined (reading 'etag')","stack":"TypeError: Cannot read properties of undefined (reading 'etag')\n    at _updateObjectExpirationDate (/usr/src/app/lib/api/apiUtils/object/coldStorage.js:194:47)\n    at Object.startRestore (/usr/src/app/lib/api/apiUtils/object/coldStorage.js:257:37)\n    at startRestore (/usr/src/app/lib/api/apiUtils/object/objectRestore.js:121:36)\n    at nextTask (/usr/src/app/node_modules/async/dist/async.js:5297:14)\n    at next (/usr/src/app/node_modules/async/dist/async.js:5304:9)\n    at /usr/src/app/node_modules/async/dist/async.js:906:16\n    at /usr/src/app/lib/api/apiUtils/object/objectRestore.js:116:28\n    at /usr/src/app/node_modules/arsenal/build/lib/s3middleware/objectRestore.js:97:16\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/arsenal/build/lib/s3middleware/objectRestore.js:5:58)","level":"fatal","message":"caught error","hostname":"fv-az661-183","pid":68}
```